src/libraries/System.Net.Sockets/src/System/Net/Sockets/ReceiveMessageOverlappedAsyncResult.Windows.cs

```diff
@@ -33,65 +33,54 @@ internal unsafe int GetSocketAddressSize()
         // since the overlapped calls may complete asynchronously.
         internal unsafe void SetUnmanagedStructures(byte[] buffer, int offset, int size, Internals.SocketAddress socketAddress, SocketFlags socketFlags)
         {
-            _messageBuffer = new byte[sizeof(Interop.Winsock.WSAMsg)];
-            _wsaBufferArray = new byte[sizeof(WSABuffer)];
+            _messageBufferPinned = GC.AllocateArray<byte>(sizeof(Interop.Winsock.WSAMsg));
```

Comment:
Do any of these need `, pinned: true` and if not is the name incorrect?

Reply:
Heh, good catch.